app/controllers/posts_controller.rb

```diff
@@ -1,7 +1,11 @@
 class PostsController < ApplicationController
   def index
-    @user = User.find(params[:user_id])
-    @posts = @user.posts.order(created_at: :desc).paginate(page: params[:page], per_page: 2)
+    # @user = User.find(params[:user_id])
+    # @posts = @user.posts.order(created_at: :desc).paginate(page: params[:page], per_page: 2)
+    @posts = Post.includes(:author)
+      .where(author_id: params[:user_id])
+      .order(created_at: :desc)
+      .paginate(page: params[:page], per_page: 2)
```

Comment:
- [ ] We are trying to eliminate N+1 query problems. Given that the purpose of eliminating this is to reduce the round trips taken to retrieve the related entities of an object, we should be eliminating the number of queries for fetching the **comments** belonging to a post in this `index` action.

  There is also no need to fetch the post author this way. We should first retrieve the author and then fetch the posts belonging to the author along with their comments. Here is a proposed code you can tweak:

  ```
  @user = User.find_by(id: params[:user_id])
  @posts = @user.posts.includes(:comments)
  ...
  ```